cypress/integration/finish_mama_project/social.spec.js

```diff
@@ -0,0 +1,191 @@
+///<reference types='cypress' />
+import { getRandomNumber } from '/cypress/support/utils.js';
+import { login } from '/cypress/support/shared.js';
+import meUser from '/cypress/fixtures/me-user.json';
+import { setJwtToken } from '/cypress/support/utils';
+
+function selectFirstArticle() {
+  waitForArticlesList();
+  cy.get('@articleList')
+    .find('.article-preview')
+    // .should('have.length', 12)
+    .should('have.length.greaterThan', 2)
+    .eq(0)
+    .as('firstArticle');
+}
+function waitForArticlesList() {
+  cy.get('@articleList')
+    .contains("div.article-preview", "Loading")
+    .should("not.be.visible");
+}
+
+function openGlobalFeed() {
+  cy.get('.feed-toggle a.nav-link[ng-class*=all]').click();
+}
+
+function checkActiveYourFeed() {
+  cy.get('@yourFeed').should('have.class', 'active');
+}
+
+function subscribeFromUser() {
+  // meUser.username
+  cy.get('@firstArticle').find('.author').as('describedAuthor').click();
+  cy.get('@describedAuthor')
+    .invoke('text')
+    .invoke('trim')
+    .as('subscribedAuthorName');
+  cy.get('follow-btn button')
+    .should('contain', 'Follow')
+    .click()
+    .as('followButton');
+  // TODO:  improve css
+  cy.get('@followButton').should('contain', 'Unfollow');
+}
+
+function unsubscribeFromUser() {
+  checkActiveYourFeed();
+  selectFirstArticle();
+  cy.get('@firstArticle').find('.author').as('undescribedAuthor').click();
+  cy.get('@undescribedAuthor')
+    .invoke('text')
+    .invoke('trim')
+    .as('unsubscribedAuthorName');
+  cy.get('follow-btn button')
+    .should('contain', 'Unfollow')
+    .click()
+    .as('UnfollowButton');
+  // TODO:  improve css
+  cy.get('@UnfollowButton').should('contain', 'Follow');
+
+}
+
+function unsubscribeFromAll() {
+  cy.get('@buttonHome').click();
+  waitForArticlesList();
+  cy.get('@articleList').find('.article-preview')
+    .then(articles => {
+      cy.log(articles.length);
+      if (articles.length > 2) {
+        cy.wrap(articles).eq(0).click();
+        unsubscribeFromUser();
+        unsubscribeFromAll();
+      }
+    });
+}
+
+function goToHomePage() {
+  cy.get('@appHeader')
+    .find('[show-authed=true] [ui-sref="app.home"]').as('buttonHome')
+    .click();
+}
+
+function deleteAllMyArticles() {
+  cy.get('@appHeader').find('[ui-sref*=username]').click();
+  debugger;
+  cy.wait(5000);
```

Comment:
Add comment why do you need it here.

Reply:
I don't understand what is wrong! But without `cy.wait(5000)` don't work. I tryed to use 
```js
// cy.location('hash').should('eq', '#/@test_manual4');
  // cy.url().should("include", "#/@test_manual4");
``` but it doesn't work
I attach video https://www.awesomescreenshot.com/video/14393554?key=918860b78fd72922afa44bb567f51de3
Help me, please